### PR TITLE
Add type hints and tests.

### PR DESCRIPTION
**Pull request by NumberPiOso**
Continuing in line with #2128.

Add type hints and doctests corrected some little changes things that are not _Python_ idiomatic.

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [X] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: searches/ternary_search.py [hunk 1]
```diff
@@ -69,35 +148,28 @@ def rec_ternary_search(left, right, A, target):
...
-
-    try:
-        __assert_sorted(collection)
-    except ValueError:
-        sys.exit("Sequence must be sorted to apply the ternary search")
-
-    target_input = input("Enter a single number to be found in the list:\n")
-    target = int(target_input)
+    user_input = input("Enter numbers separated by comma:\n").strip()
+    collection = [int(item.strip()) for item in user_input.split(",")]
```
Comment (cclauss) [inline searches/ternary_search.py:166]:
```suggestion
    collection = sorted(int(item.strip()) for item in user_input.split(","))
```
We can sort the collection and remove the assertion logic.

File: searches/ternary_search.py [hunk 2]
```diff
@@ -69,35 +148,20 @@ def rec_ternary_search(left, right, A, target):
...
+    user_input = input("Enter numbers separated by comma:\n").strip()
+    collection = sorted(int(item.strip()) for item in user_input.split(","))
+    target = int(input("Enter the number to be found in the list:\n").strip())
     result1 = ite_ternary_search(collection, target)
     result2 = rec_ternary_search(0, len(collection) - 1, collection, target)
 
-    if result2 is not None:
+    if result2 != -1:
         print(f"Iterative search: {target} found at positions: {result1}")
         print(f"Recursive search: {target} found at positions: {result2}")
```
Comment (cclauss) [inline searches/ternary_search.py:166]: Does this provide the right answer if the user inputs an out-of-order list?

Comment (NumberPiOso) [reply searches/ternary_search.py:166]: It does not, I think it is better to return to the _assert\_sorted_ logic.

Comment (cclauss) [reply searches/ternary_search.py:166]: OK but let's do it inline.  We don't need a separate function.

Comment (NumberPiOso) [reply searches/ternary_search.py:166]: I guess my error could be avoided by testing that part of the code,  however the use of inputs make it hard to test. Do you know any way that the **main** could be tested ?

Comment (cclauss) [reply searches/ternary_search.py:166]: Sure.
def perform_searches(collection: List[int], target: int) -> None:
    # put lines 159, 160, and 162-168 under this function and then you can test `perform_searches()` all day long.

Comment (NumberPiOso) [reply searches/ternary_search.py:166]: Ok, I guess thats the same than testing each one of the two searches functions.
I meant something to try to test different user inputs via the input function, not when the collection is already defined.

**Review by 53jk1 — APPROVED** [review]

**Comment by cclauss** [comment]
Please resolve conflicts.

**Comment by NumberPiOso** [comment]
Good to go ! @cclauss 

sorry for the delay.

**Comment by dhruvmanila** [comment]
Hey, if possible can you update the variable names to `snake_case` and make some of the parameter names descriptive (the ones which are of length 1).

**Comment by NumberPiOso** [comment]
I hope we are good to go.

@dhruvmanila

**Review by dhruvmanila — APPROVED** [review]
Yes! Thank you for your contribution! :)

**Pull request merged**